lightly/utils/scheduler.py

```diff
@@ -27,12 +29,9 @@ def cosine_schedule(
     if max_steps < 1:
         raise ValueError("Total step number must be >= 1")
     if step > max_steps:
-        # Note: we allow step == max_steps even though step starts at 0 and should end
-        # at max_steps - 1. This is because Pytorch Lightning updates the LR scheduler
-        # always for the next epoch, even after the last training epoch. This results in
-        # Pytorch Lightning calling the scheduler with step == max_steps.
-        raise ValueError(
-            f"The current step cannot be larger than max_steps but found step {step} and max_steps {max_steps}."
+        warnings.warn(
+            f"Current step number {step} exceeds max_steps {max_steps}.",
+            category=RuntimeWarning,
         )
```

Comment:
Changed this to warning to avoid accidentally crashing training in the very last epoch. This didn't happen in the benchmarks but could easily happen if a user passes slightly wrong arguments.